platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/datasource/VirtualLeafBytes.java

```diff
@@ -237,28 +239,45 @@ public void writeTo(final WritableSequentialData out) {
     }
 
     public int getSizeInBytesForHashing() {
-        int len = 0;
-        len += 1; // 0x00 prefix
         final Bytes kb = keyBytes();
-        len += ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_KEY, Math.toIntExact(kb.length()));
+        final int keyLen = Math.toIntExact(kb.length());
+        int innerLen = ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_KEY, keyLen);
+
         final Bytes vb = valueBytes();
-        if (vb != null) {
-            len += ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_VALUE, Math.toIntExact(vb.length()));
+        final int valueLen = (vb != null) ? Math.toIntExact(vb.length()) : -1;
+        if (valueLen >= 0) {
+            innerLen += ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_VALUE, valueLen);
         }
-        return len;
+
+        // `1 +` for 0x00 prefix
+        return 1 + ProtoWriterTools.sizeOfDelimited(FIELD_MERKLELEAF_STATEITEM, innerLen);
     }
 
     // Output size must be at least getSizeInBytesForHashing()
     public void writeToForHashing(final BufferedData out) {
         out.reset();
         assert out.remaining() >= getSizeInBytesForHashing();
+
         out.writeByte((byte) 0x00);
+
         final Bytes kb = keyBytes();
-        ProtoWriterTools.writeDelimited(out, FIELD_LEAFRECORD_KEY, Math.toIntExact(kb.length()), kb::writeTo);
+        final int keyLen = Math.toIntExact(kb.length());
+
         final Bytes vb = valueBytes();
-        if (vb != null) {
-            ProtoWriterTools.writeDelimited(out, FIELD_LEAFRECORD_VALUE, Math.toIntExact(vb.length()), vb::writeTo);
+        final int valueLen = (vb != null) ? Math.toIntExact(vb.length()) : -1;
```

Comment:
Same as my comment above in `getSizeInBytesForHashing()`

Reply:
Done